### DISCUSSION TITLE
DateTime scalar doesn't keep the offset in @DgsMutation

### DISCUSSION THREAD

**Discussion by albertogdd** [Netflix/dgs-framework#480] [Q&A] [answered]
In any given query/mutation, when we send this value for a `scalar DateTime` attribute:

```"date": "2021-07-12T11:30:00.00+02:00"```

It gets parsed to `OffsetDateTime` in the `ExtendedScalars` implementation nicely, but later on something happens before getting into `@DgsMutation`, where the date is updated to UTC:

```"date": "2021-07-12T09:30:00Z"```

This didn't happened to us with `kickstart-graphql`, is it a bug in DGS? Or are we doing something wrong

We want to validate the date offset received, but as it is now, it gets always updated to UTC automatically, hence we can't know the offset sent.

**Comment by StefanHaunsberger** [comment]
We do not have to use this DateTime scalar, we can also use our own one if that works better?

**Reply by albertogdd** [reply]
`ExtendedScalars` doesn't seem to be the issue as the `OffsetDateTime` generated there contains the correct offset.

**Comment by albertogdd** [comment]
We tested also using our own custom `DateTime` scalar and happens the same

**Comment by StefanHaunsberger** [comment]
Should we report this to the official Netflix GitHub issues?

**Reply by paulbakker** [reply]
We'll look into it, just didn't get to it yet.

**Reply by berngp** [reply]
@StefanHaunsberger is this happening with the latest version? I created a small test to assert what you mention and haven't been able to replicate. ref https://github.com/Netflix/dgs-framework/pull/506

**Comment by albertogdd** [comment, marked as answer]
@berngp Sorry for the late reply. Apparently this used to happen with earlier versions, but at some point it got fixed and now works fine, thank you for the effort!
